observationtree/src/main/scala/explore/observationtree/TargetObsList.scala

```diff
@@ -230,12 +233,27 @@ object TargetObsList {
     def render(props: Props, state: State): VdomElement = {
       val obsByTarget = props.targetsWithObs.get.obs.toList.groupBy(_.target.id)
 
+      val targets   = props.targetsWithObs.get.targets.toList
+      val targetIds = targets.map(_.id)
+
       <.div(ExploreStyles.ObsTree)(
         UndoRegion[TargetsWithObs] { undoCtx =>
           DragDropContext(onDragEnd = onDragEnd(undoCtx.setter))(
             <.div(
-              props.targetsWithObs.get.targets.elements.toTagMod { target =>
-                val targetId = target.id
+              <.div(ExploreStyles.ObsTreeButtons)(
+                <.div(
+                  Button(size = Mini, compact = true, onClick = newTarget(undoCtx.setter))(
+                    Icons.New.size(Small).fitted(true)
+                  )
+                ),
+                UndoButtons(props.targetsWithObs.get, undoCtx, size = Mini)
+              ),
+              targets.toTagMod { target =>
+                val targetId     = target.id
+                val currIdx      = targetIds.indexOf(targetId)
+                val targetIdxs   = targets.zipWithIndex
```

Comment:
Maybe this should go outside of the loop, so we only compute it once.

Reply:
How could it go outside the loop considering that we refer to `target`?

Reply:
I was referring to `val targetIdxs   = targets.zipWithIndex`